version: '3.4'

services:

  # certbot:
  #   image: certbot/certbot:latest
  #   volumes:
  #     - ./certbot/www/:/var/www/certbot/:rw
  #     - ./certbot/conf/:/etc/letsencrypt/:rw
  #     - ./certbot/conf/:/etc/nginx/ssl/:ro
  # nginx:
  #   depends_on:
  #     - api
  #     - certbot
  #   image: nginx:1.23.1-alpine
  #   restart: always
  #   volumes:
  #     - ./nginx/default.conf:/etc/nginx/conf.d/default.conf/:ro
  #     - ./certbot/www:/var/www/certbot/:ro
  #   ports:
  #     - 80:80
  #     - 443:443

  db:
    container_name: xpenddb
    image: postgres:14.2-alpine
    restart: always
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
      POSTGRES_USER: ${DB_USER}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}"]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
       - ./db-data:/var/lib/postgresql/data
  
  api:
    container_name: xpendapi
    image: rafadanam/xpendapi:latest
    depends_on:
      db:
        condition: service_healthy
    build:
      context: ./be
      args:
        APP_VERSION: ${APP_VERSION}
      target: production
    ports:
      - ${API_PORT}:5000
    environment:
      ORIGIN: ${ORIGIN}
      PORT: 5000
      DB_HOST: db
      DB_PORT: 5432
      DB_USER: ${DB_USER}
      DB_PASSWORD: ${DB_PASSWORD} 
      DB_NAME: ${DB_NAME}
      JWT_SECRET_PUBLIC: ${JWT_SECRET_PUBLIC}
      JWT_SECRET_PRIVATE: ${JWT_SECRET_PRIVATE}